en/enable-tls-between-components.md

```diff
@@ -0,0 +1,1090 @@
+---
+title: Enable TLS between TiDB Components
+summary: Learn how to enable TLS between TiDB components on Kubernetes.
+category: how-to
+---
+
+# Enable TLS between TiDB Components
+
+This document describes how to enable Transport Layer Security (TLS) between components of the TiDB cluster in Kubernetes, which is supported since TiDB Operator v1.1.
+
+To enable TLS between TiDB components, perform the following steps:
+
+1. Generate certificates for each component of the TiDB cluster to be created:
+
+   - A set of server-side certificates for the PD/TiKV/TiDB/Pump/Drainer component, saved as the Kubernetes Secret objects: `<cluster-name>-<component-name>-cluster-secret`
+   - A set of shared client-side certificates for the various clients of each component, saved as the Kubernetes Secret objects:  `<cluster-name>-cluster-client-secret`.
+
+2. Deploy the cluster, and set `.spec.tlsClient.enabled` to `true`.
+3. Configure `pd-ctl` to connect to the cluster.
+
+Certificates can be issued in multiple methods. This document describes two methods. You can choose either of them to issue certificates for the TiDB cluster:
+
+    - [Using the `cfssl` system](#using-cfssl)
+    - [Using the `cert-manager` system](#using-cert-manager)
+
+## Step 1: Generate certificates for components of the TiDB cluster
+
+This section describes how to issue certificates using two methods: `cfssl` and `cert-manager`.
+
+### Using `cfssl`
+
+1. Download `cfssl` and initialize the certificate issuer:
+
+    {{< copyable "shell-regular" >}}
+
+    ```shell
+    mkdir -p ~/bin
+    curl -s -L -o ~/bin/cfssl https://pkg.cfssl.org/R1.2/cfssl_linux-amd64
+    curl -s -L -o ~/bin/cfssljson https://pkg.cfssl.org/R1.2/cfssljson_linux-amd64
+    chmod +x ~/bin/{cfssl,cfssljson}
+    export PATH=$PATH:~/bin
+    mkdir -p ~/cfssl
+    cd ~/cfssl
+    cfssl print-defaults config > ca-config.json
+    cfssl print-defaults csr > ca-csr.json
+    ```
+
+2. Configure the client auth (CA) option in `ca-config.json`:
+
+    ```json
+    {
+        "signing": {
+            "default": {
+                "expiry": "8760h"
+            },
+            "profiles": {
+                "server": {
+                    "expiry": "8760h",
+                    "usages": [
+                        "signing",
+                        "key encipherment",
+                        "server auth",
+                        "client auth"
+                    ]
+                },
+                "client": {
+                    "expiry": "8760h",
+                    "usages": [
+                        "signing",
+                        "key encipherment",
+                        "client auth"
+                    ]
+                }
+            }
+        }
+    }
+    ```
+
+    > **Note:**
+    >
+    > Add `"client auth"` in `profile`-`server`- `usages`, because this server-side certificate is also used as the client-side certificate.
```

Comment:
```suggestion
    > Add `"client auth"` in `profiles` - `server` - `usages`, because this server-side certificate is also used as the client-side certificate.
```